apps/frontend/src/pages/project/libs/components/project-details-menu/project-details-menu.tsx

```diff
@@ -1,36 +1,51 @@
-import { useNavigate } from "react-router-dom";
-
-import { Menu, MenuItem } from "~/libs/components/components.js";
-import { AppRoute } from "~/libs/enums/enums.js";
-import { configureString } from "~/libs/helpers/helpers.js";
-import { useCallback, usePopover } from "~/libs/hooks/hooks.js";
+import { Menu, MenuItem, NavLink } from "~/libs/components/components.js";
+import { AppRoute, PermissionKey } from "~/libs/enums/enums.js";
+import { checkHasPermission, configureString } from "~/libs/helpers/helpers.js";
+import { useAppSelector, useCallback, usePopover } from "~/libs/hooks/hooks.js";
 
 type Properties = {
+	onEdit: () => void;
 	projectId: number;
 };
 
-const ProjectDetailsMenu = ({ projectId }: Properties): JSX.Element => {
+const ProjectDetailsMenu = ({
+	onEdit,
+	projectId,
+}: Properties): JSX.Element | null => {
+	const { isOpened, onClose, onOpen } = usePopover();
+	const { authenticatedUser } = useAppSelector(({ auth }) => auth);
+
+	const handleEditClick = useCallback(() => {
+		onEdit();
+		onClose();
+	}, [onEdit, onClose]);
+
+	if (!authenticatedUser) {
+		return null;
+	}
+
 	const projectAccessManagementRoute = configureString(
 		AppRoute.PROJECT_ACCESS_MANAGEMENT,
 		{
 			id: projectId.toString(),
 		},
 	);
 
-	const { isOpened, onClose, onOpen } = usePopover();
-	const navigate = useNavigate();
+	const hasManageAllProjectsPermission = checkHasPermission(
+		[PermissionKey.MANAGE_ALL_PROJECTS],
+		authenticatedUser.groups.flatMap((group) => group.permissions),
+	);
 
-	const handleManagementClick = useCallback(() => {
-		navigate(projectAccessManagementRoute);
-	}, [navigate, projectAccessManagementRoute]);
+	if (!hasManageAllProjectsPermission) {
+		return null;
+	}
 
 	return (
 		<Menu isOpened={isOpened} onClose={onClose} onOpen={onOpen}>
-			<MenuItem
-				iconName="access"
-				label="Manage Access"
-				onClick={handleManagementClick}
-			/>
+			<MenuItem iconName="pencil" label="Edit" onClick={handleEditClick} />
+			<NavLink to={projectAccessManagementRoute}>
+				<MenuItem iconName="access" label="Manage Access" />
+			</NavLink>
```

Comment:
We need to do something similar to the approach in the Button component. We need to pass onClick or href to menu item, and then render NavLink if needed inside it
```suggestion
				<MenuItem iconName="access" label="Manage Access" href={projectAccessManagementRoute} />
```

Reply:
This has now been implemented, i've made `href `and onclick optional prop for `menuItem`